src/components/molecules/UserProfileHeader.tsx

```diff
@@ -30,33 +35,92 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 interface IUserProfileHeader {
-  userDetails: IUserInfo;
+  userDetails: IAnywayUserDetails;
   isUpdateScreenOpen: boolean;
   handleLogout: () => any;
+  isAdmin: boolean;
 }
-const UserProfileHeader: React.FC<IUserProfileHeader> = ({ userDetails, isUpdateScreenOpen, handleLogout }) => {
+const UserProfileHeader: React.FC<IUserProfileHeader> = ({
+  userDetails,
+  isUpdateScreenOpen,
+  handleLogout,
+  isAdmin,
+}) => {
   const { t } = useTranslation();
-  const userData = userDetails.data;
+  const [loading, setLoading] = useState(false);
+  const store: RootStore = useStore();
+  const defaultFormInput: IFormInput = {
+    email: userDetails.data.email,
+    firstName: userDetails.data.firstName,
+    lastName: userDetails.data.lastName,
+    workplace: userDetails.data.roles.filter((role) => role !== ROLE_ADMIN_NAME)[0], // first role that is not admin
+  };
+  const saveUserChanges = async (email: string, prevOrgName: string, newOrgName: string) => {
+    if (newOrgName !== prevOrgName) {
+      try {
+        setLoading(true);
+        await store.setOrgToUser(newOrgName, email);
+        store.getUsersListInfo();
+      } catch (err) {
+        setLoading(false);
```

Comment:
use `finally` to remove loader (should be turned off on success as well)

Reply:
@yuvalbl when using "finally", the loading screen stops before the role was actually changed - it takes another extra millisecond to update the screen. that's why I used useEffect to detect when the role was actually changed, and then I change setLoading to false. 

  useEffect(() => {
    setLoading(false);
  }, [store.usersManagementTableData]);
  
  is that acceptable?